src/cordova/plugin/OutlineAndroidLib/outline/src/test/java/org/outline/vpn/VpnTunnelTest.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.outline.vpn;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class VpnTunnelTest {
+
+    @Test
+    public void constructor_rejectsNullService() {
+        assertThrows(IllegalArgumentException.class, () -> new VpnTunnel(null));
+    }
+
+}
```

Comment:
In general we end the files with new line.

Reply:
Done